nix/keys.nix

```
@@ -136,23 +136,64 @@ in
               touch /run/keys/done
             '')
         }
-      '';
 
-    systemd.services.nixops-keys =
-      { enable = config.deployment.keys != {};
-        description = "Waiting for NixOps Keys";
-        wantedBy = [ "keys.target" ];
-        before = [ "keys.target" ];
-        unitConfig.DefaultDependencies = false; # needed to prevent a cycle
-        serviceConfig.Type = "oneshot";
-        serviceConfig.RemainAfterExit = true;
-        script =
+        ${concatStringsSep "\n" (flip mapAttrsToList config.deployment.keys (name: value:
+          # Make sure each key has correct ownership, since the configured owning
+          # user or group may not have existed when first uploaded.
+          ''
+            [[ -f "/run/keys/${name}" ]] && chown '${value.user}:${value.group}' "/run/keys/${name}"
```

Maybe keep only 
```
chown '${value.user}:${value.group}' '/run/keys/${name}'
```
to fail noisily